### ISSUE TITLE
CytoTRACEKernel does not work as per API example & tutorial

### ISSUE THREAD

**Issue by sebastianbirk** [theislab/cellrank#997]
The examples given for the CytoTRACEKernel in the API documentation (https://cellrank.readthedocs.io/en/latest/api/_autosummary/kernels/cellrank.kernels.CytoTRACEKernel.html) and the tutorial (https://cellrank.readthedocs.io/en/latest/notebooks/tutorials/cellrank_meets_cytotrace.html) don't seem to work. I am using cellrank version 1.5.1 (tried installing from conda, pip and from master), am just copying the code of above examples, using them with the package internal datasets as described but a ValueError is returned "Encountered NaN values in pseudotime.". 

```python
# import packages and load data
import scvelo as scv
import cellrank as cr
adata = cr.datasets.pancreas()

# standard pre-processing
sc.pp.filter_genes(adata, min_cells=10)
sc.pp.normalize_total(adata)
sc.pp.log1p(adata)
sc.pp.highly_variable_genes(adata)

# CytoTRACE by default uses imputed data - a simple way to compute kNN-imputed data is to use scVelo's moments
# function. However, note that this function expects `spliced` counts because it's designed for RNA velocity,
# so we're using a simple hack here:
if 'spliced' not in adata.layers or 'unspliced' not in adata.layers:
    adata.layers['spliced'] = adata.X
    adata.layers['unspliced'] = adata.X

# compute kNN-imputation using scVelo's moments function
scv.pp.moments(adata)

# import and initialize the CytoTRACE kernel, compute transition matrix - done!
from cellrank.kernels import CytoTRACEKernel
ctk = CytoTRACEKernel(adata).compute_cytotrace().compute_transition_matrix()
```

```pytb
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[14], line 24
     22 # import and initialize the CytoTRACE kernel, compute transition matrix - done!
     23 from cellrank.kernels import CytoTRACEKernel
---> 24 ctk = CytoTRACEKernel(adata).compute_cytotrace().compute_transition_matrix()

File ~/miniconda3/envs/autotalker/lib/python3.9/site-packages/cellrank/tl/kernels/_cytotrace_kernel.py:95, in CytoTRACEKernel.__init__(self, adata, backward, layer, aggregation, use_raw, compute_cond_num, check_connectivity, **kwargs)
     82 def __init__(
     83     self,
     84     adata: AnnData,
   (...)
     93     **kwargs: Any,
     94 ):
---> 95     super().__init__(
     96         adata,
     97         backward=backward,
     98         time_key=Key.cytotrace("pseudotime"),
     99         compute_cond_num=compute_cond_num,
    100         check_connectivity=check_connectivity,
    101         layer=layer,
    102         aggregation=aggregation,
    103         use_raw=use_raw,
    104         **kwargs,
    105     )
    106     self._time_key = Key.cytotrace("pseudotime")

File ~/miniconda3/envs/autotalker/lib/python3.9/site-packages/cellrank/tl/kernels/_pseudotime_kernel.py:58, in PseudotimeKernel.__init__(self, adata, backward, time_key, compute_cond_num, check_connectivity, **kwargs)
     49 def __init__(
     50     self,
     51     adata: AnnData,
   (...)
     56     **kwargs: Any,
     57 ):
---> 58     super().__init__(
     59         adata,
     60         backward=backward,
     61         time_key=time_key,
     62         compute_cond_num=compute_cond_num,
     63         check_connectivity=check_connectivity,
     64         **kwargs,
     65     )
     66     self._time_key = time_key

File ~/miniconda3/envs/autotalker/lib/python3.9/site-packages/cellrank/tl/kernels/_base_kernel.py:965, in Kernel.__init__(self, adata, backward, compute_cond_num, check_connectivity, **kwargs)
    954 def __init__(
    955     self,
    956     adata: AnnData,
   (...)
    960     **kwargs: Any,
    961 ):
    962     super().__init__(
    963         adata, backward, op_name=None, compute_cond_num=compute_cond_num, **kwargs
    964     )
--> 965     self._read_from_adata(check_connectivity=check_connectivity, **kwargs)

File ~/miniconda3/envs/autotalker/lib/python3.9/site-packages/cellrank/tl/kernels/_cytotrace_kernel.py:120, in CytoTRACEKernel._read_from_adata(self, time_key, layer, aggregation, use_raw, **kwargs)
    108 def _read_from_adata(
    109     self,
    110     time_key: str,
   (...)
    116     **kwargs: Any,
    117 ) -> None:
    118     self.compute_cytotrace(layer=layer, aggregation=aggregation, use_raw=use_raw)
--> 120     super()._read_from_adata(time_key=time_key, **kwargs)

File ~/miniconda3/envs/autotalker/lib/python3.9/site-packages/cellrank/tl/kernels/_pseudotime_kernel.py:79, in PseudotimeKernel._read_from_adata(self, time_key, **kwargs)
     76     self._pseudotime = np.max(self.pseudotime) - self.pseudotime
     78 if np.any(np.isnan(self._pseudotime)):
---> 79     raise ValueError("Encountered NaN values in pseudotime.")

ValueError: Encountered NaN values in pseudotime.
```

#### Versions:

cellrank==1.5.1+g6150f02 scanpy==1.9.1 anndata==0.8.0 numpy==1.23.5 numba==0.56.4 scipy==1.9.3 pandas==1.5.3 pygpcca==1.0.4 scikit-learn==1.2.1 statsmodels==0.13.5 python-igraph==0.10.2 scvelo==0.2.5 pygam==0.8.0 matplotlib==3.7.0 seaborn==0.12.2

**Comment by Marius1311** [comment]
Hi @michalk8, we should look into this.

**Comment by maxim-h** [comment]
I'm having the same issue. 
Look like it's the same as described in  #760 which was closed via #761.  Latter was merged into `dev`, which doesn't seem to exist.
@michalk8, regarding some of the comments in #760:

- `adata.obs['ct_pseudotime']` is all NaNs.
-  `adata.X.sum(1).shape` is `(13335, 1)`, so not only HVGs.
-  `np.isinf(adata.layers['Ms']).sum()` and `np.isinf(adata.layers['Mu']).sum()` are both 0.

Any other way to troubleshoot?

**Comment by michalk8** [comment]
> Now that I installed from '1.5.1+g49dbd8fd' it runs through without error.

Glad it works and thanks for checking this! We will be releasing `2.0` within a couple of weeks.

Closing this as completed.

**Comment by davisidarta** [comment]
I'm running version `'1.5.2.dev215+gb258a4fe'` and am having the exact same error. Similar to @maxim-h,  I have already checked the AnnData object (adata) and:

* adata.obs['ct_pseudotime'] is all NaNs.
* adata.shape is (3791, 25180), so not only HVGs.
* np.isinf(adata.layers['Ms']).sum() and np.isinf(adata.layers['Mu']).sum() are both 0.

I couldn't find `'1.5.1+g49dbd8fd'` to install from. 

Any insights? Will this by any chance be fixed in upcoming 2.0?

**Comment by Marius1311** [comment]
Hi @davisidarta, yes, this will be fixed in 2.0, and should already be fixed in the latest version of the tutorial here: https://cellrank.readthedocs.io/en/latest/notebooks/tutorials/cellrank_meets_cytotrace.html

Please install CellRank from main and try running this new version of the tutorial. Let me know if the error persists.
